src/devices/MAX31856/MAX31856.cs

```diff
@@ -0,0 +1,209 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.Spi;
+using System.IO;
+using UnitsNet;
+
+namespace Iot.Device.MAX31856
+{
+    /// <summary>
+    /// Documentation https://datasheets.maximintegrated.com/en/ds/MAX31856.pdf
+    /// </summary>
+    public class MAX31856 : IDisposable
+    {
+        private readonly byte _thermocoupleType;
+        private SpiDevice _spiDevice;
+
+        #region SpiSettings
+
+        /// <summary>
+        /// Spi Clock Frequency from the Technical Manual of the device
+        /// </summary>
+        public const int SpiClockFrequency = 5_000_000;
+
+        /// <summary>
+        /// Spi Clock Frequency from the Technical Manual of the device
+        /// </summary>
+        public const int DataFlow = 0;
+
+        /// <summary>
+        /// MAX31856 SPI Mode
+        /// </summary>
+        public const SpiMode SpiMode = System.Device.Spi.SpiMode.Mode1;
+
+        #endregion
+
+        /// <summary>
+        /// Command to Get Temperature from the device
+        /// </summary>
+        /// <remarks>
+        /// Initializes the device and then reads the data next it converts the bytes to thermocouple temperature
+        /// </remarks>
+        public bool TryGetTemperature(out Temperature temperature)
+        {
+            try
+            {
+                temperature = ReadTemperature(out byte[] spiOutputData);
```

Comment:
this should be implemented reverse - the other method instead of throwing should be returning false, and GetTemperature should throw when TryGetTemperature returns false so that you avoid any throwing/catching

Reply:
If you want to preserve extra information about error then either implement it twice or have an internal overload which returns some kind of enum with Ok, ErrorWhatever1, ErrorWhatever2